internal/services/databricks/databricks_virtual_network_peerings_resource.go

```diff
@@ -0,0 +1,326 @@
+	fmtString := "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/virtualNetworks/workers-vnet"
+	virtualNetworkId := fmt.Sprintf(fmtString, id.SubscriptionId, id.ResourceGroupName)
+
+	peer.Properties.DatabricksVirtualNetwork = &vnetpeering.VirtualNetworkPeeringPropertiesFormatDatabricksVirtualNetwork{
+		Id: utils.String(virtualNetworkId),
+	}
+
+	peerMutex.Lock(resourceLock)
+	defer peerMutex.Unlock(resourceLock)
+
+	if err := pluginsdk.Retry(300*time.Second, retryDatabricksVnetPeeringsClientCreateUpdate(d, id, peer, meta)); err != nil {
+		return err
+	}
+
+	d.SetId(id.ID())
+
+	return resourceDatabricksVirtualNetworkPeeringRead(d, meta)
+}
+
+func resourceDatabricksVirtualNetworkPeeringRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).DataBricks.VnetPeeringsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := vnetpeering.ParseVirtualNetworkPeeringID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// for the import scenario, once the peering is created I can generate the workspace ID from the peering ID
+	// NOTE: this may not be right because I don't know if it is valid to create a peering with a
+	// workspace across subscription boundaries?
+	workspaceId := vnetpeering.NewWorkspaceID(id.SubscriptionId, id.ResourceGroupName, id.WorkspaceName)
+
+	resp, err := client.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("making Read request on Databricks %s: %+v", *id, err)
+	}
+
+	d.Set("resource_group_name", id.ResourceGroupName)
+	d.Set("name", id.VirtualNetworkPeeringName)
+	d.Set("workspace_id", workspaceId.ID())
+
+	if model := resp.Model; model != nil {
+		d.Set("allow_virtual_network_access", model.Properties.AllowVirtualNetworkAccess)
+		d.Set("allow_forwarded_traffic", model.Properties.AllowForwardedTraffic)
+		d.Set("allow_gateway_transit", model.Properties.AllowGatewayTransit)
+		d.Set("use_remote_gateways", model.Properties.UseRemoteGateways)
+
+		if model.Properties.DatabricksAddressSpace != nil && model.Properties.DatabricksAddressSpace.AddressPrefixes != nil {
+			d.Set("address_space_prefixes", model.Properties.DatabricksAddressSpace.AddressPrefixes)
+		}
+
+		if model.Properties.RemoteAddressSpace != nil && model.Properties.RemoteAddressSpace.AddressPrefixes != nil {
+			d.Set("remote_address_space_prefixes", model.Properties.RemoteAddressSpace.AddressPrefixes)
+		}
+
+		if model.Properties.DatabricksVirtualNetwork != nil {
+			if databricksNetwork := model.Properties.DatabricksVirtualNetwork.Id; databricksNetwork != nil {
+				d.Set("virtual_network_id", databricksNetwork)
+			}
+		}
+
+		if remoteNetwork := model.Properties.RemoteVirtualNetwork.Id; remoteNetwork != nil {
+			d.Set("remote_virtual_network_id", remoteNetwork)
+		}
```

Comment:
we should always be setting a value into the state:

```suggestion
		addressSpacePrefixes := make([]string, 0)
		if model.Properties.DatabricksAddressSpace != nil && model.Properties.DatabricksAddressSpace.AddressPrefixes != nil {
			addressSpacePrefixes = *model.Properties.DatabricksAddressSpace.AddressPrefixes
		}
		d.Set("address_space_prefixes", addressSpacePrefixes)

		remoteAddressSpacePrefixes := make([]string, 0)
		if model.Properties.RemoteAddressSpace != nil && model.Properties.RemoteAddressSpace.AddressPrefixes != nil {
			remoteAddressSpacePrefixes = *model.Properties.RemoteAddressSpace.AddressPrefixes
		}
		d.Set("remote_address_space_prefixes", remoteAddressSpacePrefixes)

		databricksVirtualNetworkId := ""
		if model.Properties.DatabricksVirtualNetwork != nil && model.Properties.DatabricksVirtualNetwork.Id != nil {
			databricksVirtualNetworkId = *model.Properties.DatabricksVirtualNetwork.Id
		}
		d.Set("virtual_network_id", databricksVirtualNetworkId)

		remoteVirtualNetworkId := ""
		if model.Properties.RemoteVirtualNetwork.Id != nil {
			remoteVirtualNetworkId = *model.Properties.RemoteVirtualNetwork.Id
		}
		d.Set("remote_virtual_network_id", remoteVirtualNetworkId)
```

Reply:
Fixed.